### PR TITLE
[API][Admin] Finish covering product scenarios

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | no                                                       |
| New feature?    | kinda                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

File: features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature [hunk 1]
```diff
@@ -17,13 +17,13 @@ Feature: Viewing details of a product with variants
         And I am logged in as an administrator
         And I am browsing products
 
-    @ui
+    @ui @no-api
```
Comment (GSadee) [inline features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature:20]: Why all of these scenarios are marked as no-api? We can probably check the proper information on GET endpoint of variant/product

File: src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php [hunk 2]
```diff
@@ -432,7 +432,7 @@ public function iShouldSeeAttributeWithValueInLocale(string $attribute, string $
     }
 
     /**
-     * @Then I should see non-translatable attribute :attribute with value :value
+     * @Then /^I should see non-translatable attribute "([^"]+)" with value ([^"]+)%$/
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php:451]: Why?

File: features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature (hunk 1 above)
Comment (NoResponseMate) [reply features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature:20]: We most likely could, but pretty much all of them (if not all) are already covered in different scenarios. Don't know if duplicating them is worth it.

File: src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php (hunk 2 above)
Comment (NoResponseMate) [reply src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php:451]: The `%` inside `:value` was behaving weirdly

File: features/product/managing_products/adding_associations_to_existing_product.feature [hunk 3]
```diff
@@ -9,10 +9,16 @@ Feature: Adding associations to an existing product
...
         And I associate as "Accessories" the "LG headphones" and "LG earphones" products
         And I save my changes
         Then I should be notified that it has been successfully edited
         And this product should have an association "Accessories" with products "LG headphones" and "LG earphones"
+
+    @api @no-ui
+    Scenario: Adding an association to na existing product
+        When I associate as "Accessories" the product "LG G3" with the products "LG headphones" and "LG earphones"
+        And this product should have an association "Accessories"
+        And this association should have products "LG headphones" and "LG earphones"
```
Comment (jakubtobiasz) [inline features/product/managing_products/adding_associations_to_existing_product.feature:24]: What was wrong with the steps above?

File: features/product/viewing_product_in_admin_panel/viewing_applied_catalog_promotions_details_within_variant.feature [hunk 4]
```diff
@@ -20,11 +20,20 @@ Feature: Seeing applied catalog promotions details within variant
...
     Scenario: Seeing applied catalog promotion details within variant
         When I access "Wyborowa Vodka" product
         Then "Wyborowa Vodka Exquisite" variant price should be decreased by catalog promotion "Winter sale" in "United States" channel
         And "Wyborowa Vodka Lemon" variant price should not be decreased by catalog promotion "Winter sale" in "United States" channel
         And "Wyborowa Vodka Exquisite" variant price should not be decreased by catalog promotion "Christmas sale" in "United States" channel
         And "Wyborowa Vodka Lemon" variant price should be decreased by catalog promotion "Christmas sale" in "United States" channel
+
+    @api @no-ui
+    Scenario: Seeing applied catalog promotion details within variant
+        When I view all variants of the product "Wyborowa Vodka"
```
Comment (jakubtobiasz) [inline features/product/viewing_product_in_admin_panel/viewing_applied_catalog_promotions_details_within_variant.feature:35]: Isn't this step valid also in the `@ui` context?

File: features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature (hunk 1 above)
Comment (jakubtobiasz) [reply features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature:20]: Meh, I feel your pain, but I guess we should cover them :/. Currently reading only this file might suggest these steps are not possible to be achieved.

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationsContext.php [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        private SectionAwareIriConverterInterface $sectionAwareIriConverter,
+        private ProductAssociationRepositoryInterface $associationRepository,
+        private SharedStorageInterface $sharedStorage,
+    ) {
+    }
+
+    /**
+     * @When /^I (associate as "[^"]+") the (product "[^"]+") with the ("[^"]+" product)$/
+     */
+    public function iAssociateAtTypeTheProductWithTheProduct(
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationsContext.php]:
```suggestion
    public function iAssociateAsTypeTheProductWithTheProduct(
```

File: features/product/managing_products/adding_associations_to_existing_product.feature (hunk 3 above)
Comment (NoResponseMate) [reply features/product/managing_products/adding_associations_to_existing_product.feature:24]: The way you add/modify an association is different in API as opposed to the UI, technically I could make this one `@ui`, but then some steps that are described in the previous one would be hidden

File: features/product/viewing_product_in_admin_panel/viewing_applied_catalog_promotions_details_within_variant.feature (hunk 4 above)
Comment (NoResponseMate) [reply features/product/viewing_product_in_admin_panel/viewing_applied_catalog_promotions_details_within_variant.feature:35]: Did you mean scenario?
There is no data regarding catalog promotions on the variants' index. That's something only visible on the product's show page.

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationsContext.php (hunk 5 above)
Comment (NoResponseMate) [reply src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationsContext.php]: How did you even catch that typo

File: features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature (hunk 1 above)
Comment (NoResponseMate) [reply features/product/viewing_product_in_admin_panel/viewing_details_of_product_with_variants.feature:20]: Added where applicable

**Review by jakubtobiasz — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @NoResponseMate!

**Pull request merged**